src/subscriptionNotifier/subscriptionNotifier.service.ts

```diff
@@ -0,0 +1,237 @@
+    const subscribers = await this.subscriptionService.getSubscribers();
+
+    subscribers?.forEach(async (subscriber) => {
+      const subscriptions = await this.subscriptionService.getSubscriptions(
+        subscriber.subscriberId,
+      );
+      if (subscriptions.length === 0) {
+        return;
+      }
+
+      const checkedAt = new Date();
+      const subscriptionsWithNews = await this.findNewsForSubscriptions(
+        subscriptions,
+      );
+      if (subscriptionsWithNews.length === 0) {
+        return;
+      }
+
+      const messages = this.createNotificationMessages(subscriptionsWithNews);
+      if (subscriber.subscriberType === 'channel') {
+        await this.sendNotificationsInThreads(subscriber, messages);
+      }
+      if (subscriber.subscriberType === 'user') {
+        await this.sendNotificationsAsDm(subscriber, messages);
+      }
+
+      await this.subscriptionService.updateNotifiedAt(subscriptions, checkedAt);
+    });
+  }
+
+  private async findNewsForSubscriptions(subscriptions: subscriptions[]) {
+    const subscriptionsWithNews = [];
+
+    for (const subscription of subscriptions) {
+      if (subscription.type === 'cabins') {
+        const news = await this.getNewCabins(subscription);
+        if (news.length > 0) {
+          subscriptionsWithNews.push({
+            subscription: subscription,
+            news: news,
+          });
+        }
+      }
+
+      if (subscription.type === 'activities') {
+        const news = await this.getNewActivities(subscription);
+        if (news.length > 0) {
+          subscriptionsWithNews.push({
+            subscription: subscription,
+            news: news,
+          });
+        }
+      }
+    }
+
+    return subscriptionsWithNews;
+  }
+
+  private async getNewCabins(subscription: subscriptions) {
+    return this.cabinService.getNewCabins(subscription.notifiedAt);
+  }
+
+  private async getNewActivities(
+    subscription: subscriptions,
+  ): Promise<activities[]> {
+    if (subscription.topic !== null) {
+      return await this.activityService.getNewActivities(
+        subscription.topic,
+        subscription.notifiedAt,
+      );
+    }
+    return [];
+  }
+
+  private createNotificationMessages(
+    subscriptionsWithNews: {
+      subscription: subscriptions;
+      news: cabins[] | activities[];
+    }[],
+  ): MessageThread[] {
+    const subscriberType = subscriptionsWithNews[0].subscription.subscriberType;
+    const today = new Date().toLocaleDateString('no-NO', {
+      day: '2-digit',
+      month: '2-digit',
+      year: 'numeric',
+    });
+    const messages = subscriptionsWithNews.map((sub) => {
+      if (sub.subscription.type === 'activities') {
+        const topic = `${sub.subscription.topic} (${today})`;
+        const mainMessage = `:point_right: ${sub.news.length} new activities that mention "${sub.subscription.topic}".\n`;
+        const embeds = sub.news.map((news) => {
+          const activity = news as activities;
```

Comment:
Can we avoid the cast here? Not a big fan of using `as` as it disables a bunch of checks in TS.

Reply:
I check before whether it is an activity:
`if (sub.subscription.type === 'activities')`
and also lower down
`if (sub.subscription.type === 'cabins')`
but if I don't cast it, I cannot access the correct fields on the news objects. They are different since they come from different tables.

Reply:
Never mind, I think I found a better solution for it without casting. :)

Reply:
Have a look at these changes. Do you find it readable? It avoids the casting and uses TS is keyword:
https://github.com/dnt-frivillig/dnt-discord-bot/commit/d06109c8fec3544e4507f8312c0dda436d748956

Reply:
Nice!  That is a better approach for handling this sort of thing :)